packages/gatsby/src/datastore/lmdb/query/common.ts

```diff
@@ -0,0 +1,141 @@
+import { DbComparator, IDbFilterStatement } from "../../common/query"
+import { IGatsbyNode } from "../../../redux/types"
+import { getValueAt } from "../../../utils/get-value-at"
+
+export function isDesc(
+  sortOrder: "asc" | "desc" | "ASC" | "DESC" | boolean | void
+): boolean {
+  return sortOrder === `desc` || sortOrder === `DESC` || sortOrder === false
+}
+
+/**
+ * Given dotted field selector (e.g. `foo.bar`) returns a plain list of values matching this selector.
+ * It is possible that the path maps to several values when one of the intermediate values is an array.
+ *
+ * Example node:
+ * {
+ *   foo: [{ bar: `bar1`}, { bar: `bar2` }]
+ * }
+ *
+ * In this case resolveFieldValue([`foo`, `bar`], node) returns [`bar1`, `bar2`]
+ *
+ * When `resolvedNodeFields` argument is passed the function first looks for values in this object
+ * and only looks in the node if the value is not found in `resolvedNodeFields`
+ */
+export function resolveFieldValue(
+  dottedFieldPath: string | Array<string>,
+  nodeOrThunk: IGatsbyNode | (() => IGatsbyNode),
+  resolvedNodeFields?: { [field: string]: unknown }
+): unknown {
+  let result
+  if (resolvedNodeFields) {
+    result = getValueAt(resolvedNodeFields, dottedFieldPath)
+  }
+  if (typeof result !== `undefined`) {
+    return result
+  }
+  const node = typeof nodeOrThunk === `function` ? nodeOrThunk() : nodeOrThunk
+  return getValueAt(node, dottedFieldPath)
+}
+
+export function shouldFilter(
+  filter: IDbFilterStatement,
+  fieldValue: unknown
+): boolean {
+  const value = filter.value
+  // FIXME: lmdb-store typing are outdated?
+  const compareKeyFn: any = compareKey
```

Comment:
is this `any` typing relevant anymore with `compareKey` being inlined and typed here?

Reply:
Nope, we can use `compareKey` directly now. Will change it.